From ab426c24c23d473596583576b1809d418ebf6258 Mon Sep 17 00:00:00 2001
From: Steve Lhomme <robux4@ycbcr.xyz>
Date: Thu, 29 Mar 2018 15:13:55 +0200
Subject: [PATCH 21/22] contrib: shout: socklen_t can be found in ws2tcpip.h on
 Windows

and make sure it's found on Windows Store builds as well
---
 contrib/src/shout/rules.mak                 |  1 +
 contrib/src/shout/shout-win32-socklen.patch | 12 ++++++++++++
 2 files changed, 13 insertions(+)
 create mode 100644 contrib/src/shout/shout-win32-socklen.patch

diff --git a/contrib/src/shout/rules.mak b/contrib/src/shout/rules.mak
index afd75f1..90cd2b5 100644
--- a/contrib/src/shout/rules.mak
+++ b/contrib/src/shout/rules.mak
@@ -26,6 +26,7 @@ libshout: libshout-$(SHOUT_VERSION).tar.gz .sum-shout
 	$(APPLY) $(SRC)/shout/shout-strings.patch
 	$(APPLY) $(SRC)/shout/shout-win32.patch
 	$(APPLY) $(SRC)/shout/shout-timeval.patch
+	$(APPLY) $(SRC)/shout/shout-win32-socklen.patch
 	$(call pkg_static,"shout.pc.in")
 	$(UPDATE_AUTOCONFIG)
 	$(MOVE)
diff --git a/contrib/src/shout/shout-win32-socklen.patch b/contrib/src/shout/shout-win32-socklen.patch
new file mode 100644
index 0000000..57821bf
--- /dev/null
+++ b/contrib/src/shout/shout-win32-socklen.patch
@@ -0,0 +1,12 @@
+--- libshout/src/common/net/sock.c.socklen	2018-03-29 15:11:28.885555300 +0200
++++ libshout/src/common/net/sock.c	2018-03-29 11:13:03.178552900 +0200
+@@ -57,6 +57,9 @@
+ #include <netdb.h>
+ #else
+ #include <winsock2.h>
++#undef WINAPI_FAMILY
++#define WINAPI_FAMILY WINAPI_FAMILY_DESKTOP_APP
++#include <ws2tcpip.h>
+ #endif
+ 
+ #include "sock.h"
-- 
2.10.1.windows.1

